src/pointer/inner.rs

```diff
@@ -22,7 +22,7 @@ mod _def {
     use super::*;
     /// The inner pointer stored inside a [`Ptr`][crate::Ptr].
     ///
-    /// `Ptr<'a, T>` is [covariant] in `'a` and `T`.
+    /// `PtrInner<'a, T>` is [covariant] in `'a` and invariant in `T`.
```

Comment:
We should probably create a test for this (and for `Ptr`). @jswrenn any ideas?

Reply:
Morally, we want the variance to be that of a &'a mut T, not &'a T, regardless of aliasing (unfortunately). So we want two tests:
 - for exclusive aliasing, we want an example that fails to compile, but would succeed if we used &T variance
 - for shared aliasing, we want an example that fails to compiled, but would succeed if we used &T variance

Reply:
Done.